lib/modules/manager/gradle/extract/consistent-versions-plugin.ts

```diff
@@ -0,0 +1,187 @@
+        const newDep: Record<string, any> = {
+          managerData: {
+            packageFile: VERSIONS_PROPS,
+            fileReplacePosition: propVerAndPos.filePos,
+          },
+          depName: exactDep,
+          currentValue: propVerAndPos.version,
+          currentVersion: propVerAndPos.version,
+          lockedVersion: exactVer,
+          groupName: regexToGroupString(propDepRegEx),
+        } as PackageDependency<GradleManagerData>;
+        extractedDeps.push(newDep);
+      }
+    }
+  }
+  return extractedDeps;
+}
+
+//----------------------------------
+// Private utility functions below
+//----------------------------------
+
+// Translate glob syntax to a regex that does the same
+function globToRegex(depName: string): RegExp {
+  return regEx(
+    depName
+      .replaceAll('*', '_WC_CHAR_')
+      .replace(/[/\-\\^$*+?.()|[\]{}]/g, '\\$&')
+      .replaceAll('_WC_CHAR_', '[^:]*')
+  );
+}
+
+// Translate the regex from versions.props into a group name used in branch names etc
+function regexToGroupString(regExp: RegExp): string {
+  return regExp.source.replaceAll('[^:]*', '-').replaceAll('\\.', '.');
+}
+
+interface VersionWithPosition {
+  version?: string;
+  filePos?: number;
+}
+
+/**
+ * Parses `versions.lock`
+ */
+function parseLockFile(input: string): Map<string, string> {
+  const lockLineRegex = regEx(
+    `^(?<depName>[^:]+:[^:]+):(?<lockVersion>[^ ]+) \\(\\d+ constraints: [0-9a-f]+\\)$`
+  );
+
+  const depVerMap = new Map<string, string>();
+  for (const line of input.split(newlineRegex)) {
+    const lineMatch = lockLineRegex.exec(line);
+    if (lineMatch?.groups) {
+      const { depName, lockVersion } = lineMatch.groups;
+      depVerMap.set(depName, lockVersion);
+    }
+  }
+  logger.trace(
+    `Found ${depVerMap.size} locked dependencies in ${VERSIONS_LOCK}.`
+  );
+  return depVerMap;
+}
+
+/**
+ * Parses `versions.props`
+ */
+function parsePropsFile(
+  input: string
+): [Map<string, VersionWithPosition>, Map<RegExp, VersionWithPosition>] {
+  const propsLineRegex = regEx(
+    `^(?<depName>[^:]+:[^=]+?) *= *(?<propsVersion>.*)$`
+  );
+  const depVerExactMap = new Map<string, VersionWithPosition>();
+  const depVerRegexMap = new Map<RegExp, VersionWithPosition>();
+
+  let startOfLineIdx = 0;
+  for (const line of input.split(newlineRegex)) {
+    const lineMatch = propsLineRegex.exec(line);
+    if (lineMatch?.groups) {
+      const { depName, propsVersion } = lineMatch.groups;
+      const startPosInLine = line.lastIndexOf(propsVersion);
+      const propVersionPos = startOfLineIdx + startPosInLine;
+      if (depName.includes('*')) {
+        depVerRegexMap.set(globToRegex(depName), {
+          version: propsVersion,
+          filePos: propVersionPos,
+        });
+      } else {
+        depVerExactMap.set(depName, {
+          version: propsVersion,
+          filePos: propVersionPos,
+        });
+      }
+    }
+    startOfLineIdx += line.length + 1;
```

Comment:
\+ 1 won't work with `\r\n` line endings

1) Probably do a similar check as dockerfile manager
2) Please add a test for CRLF as well that ensures that `fileReplacePosition` is set correctly in such a case

Reply:
Fixed, and with test

Reply:
Please read this:
https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#resolve-review-comments-instead-of-commenting

🙏

Reply:
Sorry. Different practices on who resolves conversations, but I can resove the ones I feel I'm done with and then @Churro can re-open if he feels it is not yet done :)